spec/system/document_collection_spec.rb

```diff
@@ -67,62 +67,57 @@
         expect(page).not_to have_selector("nav a", text: "Empty Group")
       end
     end
-  end
 
-  # test "renders no contents list if body has no h2s and is long and collection groups are empty" do
-  #   content_item = get_content_example("document_collection")
-
-  #   content_item["details"]["body"] = <<~HTML
-  #     <div class="empty group">
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #     </div>
-  #   HTML
-
-  #   content_item["details"]["collection_groups"] = [
-  #     {
-  #       "body" => "<div class=\"empty group\">\n</div>",
-  #       "documents" => [],
-  #       "title" => "Empty Group",
-  #     },
-  #   ]
-
-  #   content_item["base_path"] += "-no-h2s"
-
-  #   stub_content_store_has_item(content_item["base_path"], content_item.to_json)
-  #   visit(content_item["base_path"])
-  #   assert_not page.has_css?(".gem-c-contents-list")
-  # end
+    context "when all document collection groups are empty" do
+      context "when no headers are supplied" do
+        let(:content_item) do
+          GovukSchemas::Example.find(:document_collection, example_name: "document_collection").tap do |item|
+            item["details"]["headers"] = []
+
+            item["details"]["collection_groups"] = [
+              {
+                "body" => "<div class=\"empty group\">\n</div>",
+                "documents" => [],
+                "title" => "Empty Group",
+              },
+            ]
+          end
+        end
 
-  # test "renders contents list if body has h2s and collection groups are empty" do
-  #   content_item = get_content_example("document_collection")
-
-  #   content_item["details"]["body"] = <<~HTML
-  #     <div class="empty group">
-  #       <h2 id="one">One</h2>
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #       <h2 id="two">Two</h2>
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #       <h2 id="three">Three</h2>
-  #       <p>#{Faker::Lorem.characters(number: 200)}</p>
-  #     </div>
-  #   HTML
-
-  #   content_item["details"]["collection_groups"] = [
-  #     {
-  #       "body" => "<div class=\"empty group\">\n</div>",
-  #       "documents" => [],
-  #       "title" => "Empty Group",
-  #     },
-  #   ]
-
-  #   content_item["base_path"] += "-h2s"
-
-  #   stub_content_store_has_item(content_item["base_path"], content_item.to_json)
-  #   visit(content_item["base_path"])
-  #   assert page.has_css?(".gem-c-contents-list")
-  # end
+        it "does not render a contents list" do
+          visit base_path
+
+          expect(page).not_to have_selector(".gem-c-contents-list")
+        end
+      end
+
+      context "when h2 headers are supplied" do
```

Comment:
I think we should also [test that nested headings aren't included in the contents list](https://github.com/alphagov/frontend/blob/main/spec/presenter/corporate_information_page_presenter_spec.rb#L15-L38), as Whitehall will automatically extract all headings from the body and will be shown unless [specifically removed](https://github.com/alphagov/frontend/blob/main/app/presenters/corporate_information_page_presenter.rb#L43-L47).

Reply:
It was tested implicitly, in fact, because there are L3 headers in the document_collection_with_body example, but I've made it more explicit https://github.com/alphagov/frontend/pull/4839/files#diff-50919a8b21f824666a72e806f02b0ff691f7050bc5064142d7aa5aba1f01c931R106-R119 (as